openbb_platform/providers/fred/openbb_fred/models/bond_indices.py

```diff
@@ -0,0 +1,590 @@
+                and "," in values["index"]
+            ):
+                message = "Multiple indices not allowed for: 'yield_curve'."
+                messages.append(message)
+        if is_yield_curve is False:
+            indices = (
+                values["index"]
+                if isinstance(values["index"], list)
+                else values["index"].split(",")
+            )
+            for index in indices:
+                if values["category"] == "us":
+                    if index not in BAML_CATEGORIES.get("us"):
+                        message = (
+                            "Invalid index for category: 'us'."
+                            + f" Must be one of {','.join(BAML_CATEGORIES.get('us'))}."
+                        )
+                        messages.append(message)
+                    else:
+                        new_index.append(index)
+                if values["category"] == "high_yield":
+                    if index not in ("us", "europe", "emerging"):
+                        message = (
+                            "Invalid index for category: 'high_yield'."
+                            + f" Must be one of {','.join(BAML_CATEGORIES.get('high_yield', ''))}."
+                        )
+                        messages.append(message)
+                    else:
+                        new_index.append(index)
+                if values["category"] == "emerging_markets":
+                    if index not in BAML_CATEGORIES.get("emerging_markets"):
+                        message = (
+                            "Invalid index for category: 'emerging_markets'."
+                            + f" Must be one of {','.join(BAML_CATEGORIES.get('emerging_markets', ''))}."
+                        )
+                        messages.append(message)
+                    else:
+                        new_index.append(index)
+        if not new_index:
+            raise ValueError(
+                "No valid combinations of parameters were found."
+                + f"\n{','.join(messages) if messages else ''}"
+            )
+        if messages:
+            warn(",".join(messages))
+
+        symbols: List = []
+        if "yield_curve" in values["index"]:
+            maturities_dict = BAML_CATEGORIES[values["category"]][values["index"]]
+            maturities = list(maturities_dict)
+            symbols = [
+                maturities_dict[item][values["index_type"]] for item in maturities
+            ]
+        else:
+            items = (
+                values["index"]
+                if isinstance(values["index"], list)
+                else values["index"].split(",")
+            )
+            symbols = [
+                BAML_CATEGORIES[values["category"]]
+                .get(item, {})
+                .get(values["index_type"])
+                for item in items
+            ]
+            symbols = [symbol for symbol in symbols if symbol]
+        if not symbols:
+            raise ValueError(
```

Comment:
If these error messages are for the user it should be raising `OpenBBError` instead.

`ValueError` (or any python base exception) will display `Unexpected Error` in the UI